File: topicos/poo/node/ex2/Leitor.js
export default class Leitor {
  idade;
  nome;
  livro;


  constructor(idade, nome, livro) {
    this.idade = idade;
    this.nome = nome;
    this.setLivro(livro);
  }


  oqueEstaLendo() {
    console.log(`${this.nome} esta lendo ${this.livro.titulo} escrito por ${this.livro.autor}`);
  }

  setLivro(livro) {
    this.livro = livro;
    this.livro.incluirLeitor(this);
  }
}

File: topicos/poo/go/ex1/leitor.go
package biblioteca

import "fmt"

type Leitor struct {
	Idade      int
	Nome       string
	LivroAtual Livro
}

func NewLeitor(idade int, nome string, livro Livro) *Leitor {
	return &Leitor{
		Idade:      idade,
		Nome:       nome,
		LivroAtual: livro,
	}
}

func (l Leitor) novoLivro(livro Livro) {
	l.LivroAtual = livro
}

func (l Leitor) OqueEstaLendo() {
	fmt.Printf("%s esta lendo: %s, do autor %s", l.Nome, l.LivroAtual.Titulo, l.LivroAtual.Autor)
}
